driver-core/src/main/java/com/datastax/driver/core/FunctionMetadata.java

```diff
@@ -211,6 +204,13 @@ private String asCQLQuery(boolean formatted) {
         return sb.toString();
     }
 
+    // signatures types must not contain the keyword frozen
+    static String unfreeze(String typeStr) {
```

Comment:
This is a bit clumsy but works; the goal is to remove the frozen keyword when it is the outermost piece of the whole CQL type literal.